### PR TITLE
version: use git to determine version number

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

**Comment by jonboulle** [comment]
Eyeballing it looks fine but you made travis sad.

**Comment by crawford** [comment]
Yeah, travis doesn't copy the git tree into its environment.

File: build [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ORG_PATH="github.com/coreos"
 REPO_PATH="${ORG_PATH}/fleet"
+VERSION=$(git describe --dirty)
+GLDFLAGS="-X github.com/coreos/fleet/version.Version \"${VERSION}\""
```
Comment (jonboulle) [inline build]: where did we end up on the space vs equals? https://github.com/coreos/ignition/blob/master/build#L7

Comment (crawford) [inline build]: I'll leave the space as long as we are testing with 1.4.

**Comment by crawford** [comment]
My workaround is to set the default version to `0.0.0`. Thoughts?

**Comment by jonboulle** [comment]
0.0.0+wasnotbuiltproperly :-)

**Comment by crawford** [comment]
Done!

**Comment by crawford** [comment]
Fixes https://github.com/coreos/fleet/issues/1377.

**Comment by jonboulle** [comment]
Thanks buddy.

**Pull request merged**